src/components/P2P/index.ts

```diff
@@ -551,7 +551,7 @@ export class OceanP2P extends EventEmitter {
         `Trying to store and advertise ${list.length} initial DDOS`,
         true
       )
-      const db = this.getDatabase().ddo
+      const db = this.db.ddo
```

Comment:
what if the instance of p2p was created without specifying the db param since it is optional ? this.db will be undefined and this will lead to the clasic cannout read property of undefined 😃

Reply:
That is only called on `index.ts` :

```
if (config.hasIndexer && dbconn) {
  indexer = new OceanIndexer(dbconn, config.supportedNetworks)
  // if we set this var
  // it also loads initial data (useful for testing, or we might actually want to have a bootstrap list)
  // store and advertise DDOs
  if (process.env.LOAD_INITIAL_DDOS) {
    const list = loadInitialDDOS()
    if (list.length > 0) {
      // we need a timeout here, otherwise we have no peers available
      setTimeout(() => {
        node.storeAndAdvertiseDDOS(list)
      }, 3000)
    }
  }
}
```
there are a couple of checks there as you can see :-) , and this is something just for testing & loading initial DDOS.. so that is not a problem at all
but i can make an extra "if" for sure... just to avoid misuse

Reply:
Maybe just add `?` as an extra check to avoid anything going wrong 

```suggestion
      const db = this.db?.ddo
```

Reply:
just added the extra if, no big deal..